### PR TITLE
refactor(model): sync whisper with SupportsTranscription interface

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

- This PR implements classmethods for transcription to sync interface.
https://github.com/vllm-project/vllm/blob/v0.22.0/docs/contributing/model/transcription.md
- It enables to detect language automatically.
- pytest,...

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: vllm_rbln/model_executor/models/optimum/whisper.py [hunk 1]
```diff
@@ -40,23 +50,138 @@ class RBLNOptimumWhisperForConditionalGeneration(
...
+                "from audio. To skip detection, pass the `language` field "
                 "in the TranscriptionRequest."
             )
-            language = "en"
+            return None
         return super().validate_language(language)
 
-    # FIXME more method needed
+    @classmethod
+    def get_generation_prompt(
```
Comment (rebel-jonghewk) [inline vllm_rbln/model_executor/models/optimum/whisper.py]: @rebel-eunji I think there class methods are almost identical to upstream version. Can we reuse it instead of copy paste?

Comment (rebel-eunji) [reply vllm_rbln/model_executor/models/optimum/whisper.py]: This approach requires inheriting from `WhisperForConditionalGeneration` in upstream vLLM. Are we okay with that?

Comment (rebel-eunji) [reply vllm_rbln/model_executor/models/optimum/whisper.py]: It works with https://github.com/RBLN-SW/vllm-rbln/pull/668/changes/4c1a6e4a7c081cde954bf37acde27a25be32da3f

File: vllm_rbln/model_executor/models/optimum/whisper.py [hunk 2]
```diff
@@ -14,50 +14,21 @@
...
 
-logger = init_logger(__name__)
-
 
 class RBLNOptimumWhisperForConditionalGeneration(
     RBLNOptimumModelBase,
     RBLNOptimumDecoderMixin,
-    SupportsTranscription,
-    SupportsMultiModal,
+    WhisperForConditionalGeneration,
```
Comment (rebel-jonghewk) [inline vllm_rbln/model_executor/models/optimum/whisper.py:30]: when inheriting, since `supports_lora=True` is the default, don't we have to add supports_lora=false ?

File: vllm_rbln/model_executor/models/optimum/whisper.py [hunk 3]
```diff
@@ -14,50 +14,21 @@
...
 ):
-    # Whisper only supports audio-conditioned generation.
-    supports_transcription_only = True
-    supports_segment_timestamp = True
-    supported_languages = ISO639_1_SUPPORTED_LANGS
-
-    @classmethod
-    def validate_language(cls, language: str | None) -> str | None:
```
Comment (rebel-jonghewk) [inline vllm_rbln/model_executor/models/optimum/whisper.py:46]: Can you add a code (pytest) that `validate_language` works?

File: vllm_rbln/model_executor/models/optimum/whisper.py (hunk 2 above)
Comment (rebel-eunji) [reply vllm_rbln/model_executor/models/optimum/whisper.py:30]: It is not working. I think I should raise an error when lora_config is enabled.
Which one is better?
A) revert the inheritance and implemented the required classmethods
B) inherit WhisperForConditionalGeneration and raise an error in case of lora_config is enabled.

Comment (rebel-jonghewk) [reply vllm_rbln/model_executor/models/optimum/whisper.py:30]: b for now and add todo?

File: vllm_rbln/model_executor/models/optimum/whisper.py (hunk 3 above)
Comment (rebel-eunji) [reply vllm_rbln/model_executor/models/optimum/whisper.py:46]: https://github.com/RBLN-SW/vllm-rbln/pull/668/changes/957396e23655843055a2d47d5a95f59dfeb47879

File: vllm_rbln/model_executor/models/optimum/whisper.py (hunk 2 above)
Comment (rebel-eunji) [reply vllm_rbln/model_executor/models/optimum/whisper.py:30]: Adding log is not difficult. So I've added it.
https://github.com/RBLN-SW/vllm-rbln/pull/668/changes/e27475e9fab70b453b6455b2b90fc831cbc3dc5d

**Review by rebel-jonghewk — APPROVED** [review]
